npm-global-without-sudo.md

```diff
@@ -21,13 +21,28 @@ npm config set prefix "${HOME}/.npm-packages"
 Add the following to your `.bashrc`/`.zshrc`:
 
 ```sh
-NPM_PACKAGES="${HOME}/.npm-packages"
+NPM_PACKAGES="${XDG_DATA_HOME}/npm-packages"
 
-export PATH="$PATH:$NPM_PACKAGES/bin"
+if ! [ -d "$NPM_PACKAGES" ] && command -v npm >/dev/null 2>&1; then
+	npm config set prefix "$NPM_PACKAGES"
+	npm config set cache "${XDG_CACHE_HOME}/npm"
+fi
+
+if [[ ":${PATH}:" != *":${NPM_PACKAGES}/bin:"* ]]; then
+	export PATH="${PATH:+"${PATH}:"}${NPM_PACKAGES}/bin"
+fi
 
 # Preserve MANPATH if you already defined it somewhere in your config.
 # Otherwise, fall back to `manpath` so we can inherit from `/etc/manpath`.
-export MANPATH="${MANPATH-$(manpath)}:$NPM_PACKAGES/share/man"
+[ -z "$MANPATH" ] && export MANPATH="$(manpath)"
+if [[ ":${MANPATH}:" != *":${NPM_PACKAGES}/share/man:"* ]]; then
+	export MANPATH="${MANPATH:+"${MANPATH}:"}${NPM_PACKAGES}/share/man"
+fi
+
+# From https://github.com/npm/npm/issues/6675#issuecomment-168053009
+export NPM_CONFIG_USERCONFIG="$XDG_CONFIG_HOME/.npmrc"
```

Comment:
Same here. macOS support.

Reply:
The existence of `$MANPATH` is ensured by the above `[ -z "$MANPATH" ] && export MANPATH="$(manpath)"`, isn't it?

Reply:
I'm not sure how this is related to `$MANPATH`? You're overriding the location of the global `.npmrc` file and you need to support `$HOME` to for Mac users.

Reply:
Okay, the latest commit addreses this.